### DISCUSSION TITLE
Modules/compiler updates on Hera

### DISCUSSION THREAD

**Discussion by cristianastan2** [NOAA-EMC/global-workflow#2433] [Q&A] [answered]
Hi,

After the recent upgrades on Hera the modules invoked by 'module_base.hera.lua' are no longer available. I updated module_base.hera.lua with the new version available under the develop branch of the global workflow along with the 'load_fv3gfs_modules.sh'. I added the versions/ directory, the 'detect_machine.sh ' and 'module-setup.sh'. The new 'load_fv3gfs_modules.sh' is looking for a file versions/run.ver, which is not in the versions/ directory. Should I change <source "${HOMEgfs}/versions/run.ver"> to source "${HOMEgfs}/versions/run.hera.ver", which is available in the versions/ directory. 

Thank you, Cristiana

**Comment by KateFriedman-NOAA** [comment]
@cristianastan2 If you've run the link script step (`sorc/link_workflow.sh`) then the machine-specific `run.MACHINE.ver` will be copied into `run.ver` (e.g. `run.hera.ver` -> `run.ver`). Double check you've run that setup step under the `/sorc` folder. See more on that step here: https://global-workflow.readthedocs.io/en/latest/clone.html#link-components

Hera is undergoing a transition to Rocky8 from CentOS. The global-workflow does not yet support Rocky8. If you need to run on Hera right now make sure you are on a CentOS login node (hfe01-hfe04 currently). We will be completing our port to Rocky8 on Hera shortly.

Hope the above helps! If you have issues with global-workflow on Hera with CentOS let us know. Stay tuned for the Rocky8 update.

**Reply by cristianastan2** [reply]
Use a very old version of the g-w, which does not have `sorc/link_workflow.sh`. I copied manually the file and it worked. The `preamble.sh` is now causing some problems. I replaced `preamble.sh ` with the new version but it didn't help. I still get the same error.

/scratch2/NCEPDEV/marine/Cristiana.Stan/p8_climate/expdir/20150801/config.base: line 92: NETCDF: unbound variable

I will try to setup the experiment again and hopefully that will help.

**Reply by KateFriedman-NOAA** [reply]
@cristianastan2 The `link_workflow.sh` script has been in global-workflow for many years, so you should discontinue using such an old version. It is best to await the Rocky8 port completion on Hera and then begin using the updated `develop` branch. Versions older than 6 months are not supported.

**Comment by aerorahul** [comment]
Hera has been ported to Rocky for a while.  Is this discussion still relevant?

**Comment by cristianastan2** [comment]
Hi Rahul,

Yes, Hera has been ported to Rocky but the version of the global workflow that I was working with is not working. The policy that versions older than 6 months are not supported is very discouraging.

Cristiana


________________________________

**Reply by aerorahul** [reply, marked as answer]
The amount of work to port a version/hash of global-workflow pre-Rocky8 is unattainable with the limited resources available to the development team.
